### PR TITLE
adding code

### PR DESCRIPTION
**Pull request by stasevich741**
add files

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,16 +1,95 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
 
-    <properties>
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
-    </properties>
+  <groupId>org.example</groupId>
+  <artifactId>development</artifactId>
```
Comment (Maksim-maevsky) [inline pom.xml]: pls, read about artifact, i gues it is not good naming

File: pom.xml [hunk 2]
```diff
@@ -1,16 +1,95 @@
...
+  <version>1.0-SNAPSHOT</version>
+  <packaging>war</packaging>
```
Comment (Maksim-maevsky) [inline pom.xml]: use jar

File: pom.xml [hunk 3]
```diff
@@ -1,16 +1,95 @@
...
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
```
Comment (Maksim-maevsky) [inline pom.xml]: move dependencies to properties

File: pom.xml [hunk 4]
```diff
@@ -1,16 +1,95 @@
...
+      <scope>test</scope>
+    </dependency>
+
+    <!-- https://mvnrepository.com/artifact/org.postgresql/postgresql -->
+    <dependency>
+      <groupId>org.postgresql</groupId>
+      <artifactId>postgresql</artifactId>
+      <version>42.3.3</version>
```
Comment (Maksim-maevsky) [inline pom.xml]: move dependencies to properties

File: pom.xml [hunk 5]
```diff
@@ -1,16 +1,95 @@
...
+    </dependency>
+
+    <!-- https://mvnrepository.com/artifact/org.projectlombok/lombok -->
+    <dependency>
+      <groupId>org.projectlombok</groupId>
+      <artifactId>lombok</artifactId>
+      <version>1.18.20</version>
```
Comment (Maksim-maevsky) [inline pom.xml]: move dependencies to properties

File: pom.xml [hunk 6]
```diff
@@ -1,16 +1,95 @@
...
+
+  </dependencies>
+
+  <build>
+    <finalName>development</finalName>
+    <pluginManagement><!-- lock down plugins versions to avoid using Maven defaults (may be moved to parent pom) -->
+      <plugins>
+        <plugin>
+          <artifactId>maven-clean-plugin</artifactId>
+          <version>3.1.0</version>
```
Comment (Maksim-maevsky) [inline pom.xml]: move dependencies to properties

File: src/main/java/cinema/dao/Dao.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    List<T> findAll();
+
+    Optional<T> findById(Long id);
+
+    void delete(Long id);
+
+    void update(T entity);
+
+    void save(T entity);
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/dao/Dao.java]: may be better get T instead void

File: src/main/java/cinema/dao/FilmDao.java [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+import java.util.List;
+import java.util.Optional;
+
+public class FilmDao implements Dao<Film> {
+
+    private static final String FIND_ALL_FILMS_QUERY = "select id, name, date_and_time, tickets_list from film";
+    private static final String FIND_BY_ID_FILM_QUERY = "select id, name, date_and_time, tickets_list from film where id=?";
+    private static final String DELETE_FILM_BY_ID_QUERY = "delete from film where id=?";
+    private static final String INSERT_FILM_QUERY = "insert into film ( name, date_and_time, tickets_list ) VALUES (?,?,?)";
+    private static final String UPDATE_FILM_QUERY = "update film set name =?,date_and_time=?,tickets_list=? where id=?";
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/dao/FilmDao.java]: what do you  think about speсial class for queries?

File: src/main/java/cinema/entity/Film.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package cinema.entity;
+
+import lombok.Value;
+
+import java.time.LocalDateTime;
+
+@Value
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/Film.java]: Why do you use @Value annotation here?

File: src/main/java/cinema/entity/Ticket.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package cinema.entity;
+
+import lombok.Value;
+
+import java.math.BigDecimal;
+
+@Value
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/Ticket.java]: Why do you use https://github.com/value annotation here?

File: src/main/java/cinema/util/PropertiesUtil.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+import cinema.exception.DaoException;
+import lombok.experimental.UtilityClass;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+@UtilityClass
+public final class PropertiesUtil {
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/util/PropertiesUtil.java:9]: why do you write final if you use @UtilityClass.

Comment (stasevich741) [reply src/main/java/cinema/util/PropertiesUtil.java:9]: deleted final keyword

File: src/main/java/cinema/entity/Ticket.java (hunk 10 above)
Comment (stasevich741) [reply src/main/java/cinema/entity/Ticket.java]: replaced with @data annotation

File: src/main/java/cinema/entity/Film.java (hunk 9 above)
Comment (stasevich741) [reply src/main/java/cinema/entity/Film.java]: replaced with @data annotation

File: src/main/java/cinema/dao/FilmDao.java (hunk 8 above)
Comment (stasevich741) [reply src/main/java/cinema/dao/FilmDao.java]: added three classes FilmQuery, TicketQuery and UserQuery

File: src/main/java/cinema/dao/Dao.java (hunk 7 above)
Comment (stasevich741) [reply src/main/java/cinema/dao/Dao.java]: fixed

File: pom.xml (hunk 2 above)
Comment (stasevich741) [reply pom.xml]: fixed

File: pom.xml (hunk 1 above)
Comment (stasevich741) [reply pom.xml]: i changed like this
```
 <groupId>by.itechart</groupId>
  <artifactId>cimena-project</artifactId>
```
is it ok?

File: pom.xml (hunk 4 above)
Comment (stasevich741) [reply pom.xml]: moved

File: pom.xml (hunk 3 above)
Comment (stasevich741) [reply pom.xml]: moved

File: pom.xml (hunk 5 above)
Comment (stasevich741) [reply pom.xml]: moved

File: pom.xml (hunk 6 above)
Comment (stasevich741) [reply pom.xml]: moved

File: pom.xml (hunk 2 above)
Comment (stasevich741) [reply pom.xml]: <packaging>jar</packaging>

Comment (stasevich741) [reply pom.xml]: `<packaging>jar</packaging>`

File: pom.xml (hunk 3 above)
Comment (stasevich741) [reply pom.xml]:
```
 <properties>
//
 <!--        variables for dependencies-->
        <postges.version>42.3.3</postges.version>
        <junit.version>4.11</junit.version>
        <lombok.version>1.18.20</lombok.version>
    </properties>

 <dependency>
            <groupId>org.postgresql</groupId>
            <artifactId>postgresql</artifactId>
            <version>${postges.version}</version>
        </dependency>
```

File: src/main/java/cinema/dao/Dao.java (hunk 7 above)
Comment (stasevich741) [reply src/main/java/cinema/dao/Dao.java]: T update(T entity);

    T save(T entity);

File: src/main/java/cinema/dao/FilmDao.java (hunk 8 above)
Comment (stasevich741) [reply src/main/java/cinema/dao/FilmDao.java]:
```
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class FilmQuery {

    public static final String FIND_ALL_FILMS_QUERY = "select id, name, date_and_time, tickets_list from film";
    public static final String FIND_BY_ID_FILM_QUERY = "select id, name, date_and_time, tickets_list from film where id=?";
    public static final String DELETE_FILM_BY_ID_QUERY = "delete from film where id=?";
    public static final String INSERT_FILM_QUERY = "insert into film ( name, date_and_time, tickets_list ) VALUES (?,?,?)";
    public static final String UPDATE_FILM_QUERY = "update film set name =?,date_and_time=?,tickets_list=? where id=?";
}

```
added in class FilmDao
`import static cinema.query.FilmQuery.*;`

File: pom.xml (hunk 1 above)
Comment (stasevich741) [reply pom.xml]: or i should name groupId **ITechArt-Internship-Cinema**?

File: pom.xml [hunk 12]
```diff
@@ -1,16 +1,95 @@
...
+    <finalName>cinema_project</finalName>
+    <pluginManagement><!-- lock down plugins versions to avoid using Maven defaults (may be moved to parent pom) -->
+      <plugins>
+        <plugin>
+          <artifactId>maven-clean-plugin</artifactId>
+          <version>3.1.0</version>
+        </plugin>
+        <!-- see http://maven.apache.org/ref/current/maven-core/default-bindings.html#Plugin_bindings_for_war_packaging -->
+        <plugin>
+          <artifactId>maven-resources-plugin</artifactId>
```
Comment (Maksim-maevsky) [inline pom.xml]: do ou realy need all of these plugins?

Comment (stasevich741) [reply pom.xml]: i really dont know) ....
delete this one - ` <artifactId>maven-resources-plugin</artifactId>` or delete all of them?

Comment (stasevich741) [reply pom.xml]: deleted

File: src/main/java/cinema/entity/User.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class User {
+
+    Long id;
+    String login;
+    String password;
+    Role userRole;
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/User.java]: why is it not private? and others entities and dtos

File: src/main/java/cinema/query/TicketQuery.java [hunk 14]
```diff
@@ -0,0 +1,14 @@
+package cinema.query;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TicketQuery {
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/query/TicketQuery.java]: why not final class?

File: src/main/java/cinema/query/UserQuery.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package cinema.query;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class UserQuery {
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/query/UserQuery.java]: why not final class?

File: src/main/java/cinema/util/PropertiesUtil.java [hunk 16]
```diff
@@ -8,7 +8,7 @@
 import java.util.Properties;
 
 @UtilityClass
-public final class PropertiesUtil {
+public class PropertiesUtil {
 
     private static final Properties PROPERTIES = new Properties();
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/util/PropertiesUtil.java:11]: why static if you use @UtilityClass?

File: src/main/java/cinema/entity/User.java (hunk 13 above)
Comment (stasevich741) [reply src/main/java/cinema/entity/User.java]: fixed

File: src/main/java/cinema/query/TicketQuery.java (hunk 14 above)
Comment (stasevich741) [reply src/main/java/cinema/query/TicketQuery.java]: fixed

File: src/main/java/cinema/query/UserQuery.java (hunk 15 above)
Comment (stasevich741) [reply src/main/java/cinema/query/UserQuery.java]: fixed

File: src/main/java/cinema/util/PropertiesUtil.java (hunk 16 above)
Comment (stasevich741) [reply src/main/java/cinema/util/PropertiesUtil.java:11]: fixed

File: src/main/java/cinema/TestDao.java [hunk 17]
```diff
@@ -0,0 +1,7 @@
+package cinema;
+
+public class TestDao {
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/TestDao.java]: is it test class?

File: src/main/java/cinema/entity/Ticket.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.*;
+
+import java.math.BigDecimal;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/Ticket.java]: @Data include @Getter and @Setter

File: src/main/java/cinema/entity/User.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
+package cinema.entity;
+
+import lombok.*;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/User.java]: @Data include @Getter and @Setter

File: src/main/java/cinema/util/ConnectionManager.java [hunk 20]
```diff
@@ -0,0 +1,37 @@
+package cinema.util;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class ConnectionManager {
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/util/ConnectionManager.java]: is it final class?

File: src/main/java/cinema/entity/Film.java [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+
+import lombok.*;
+
+import java.time.LocalDateTime;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
```
Comment (Maksim-maevsky) [inline src/main/java/cinema/entity/Film.java]: @Data include @Getter and @Setter

File: src/main/java/cinema/TestDao.java (hunk 17 above)
Comment (stasevich741) [reply src/main/java/cinema/TestDao.java]: renamed to StartApp

File: src/main/java/cinema/entity/Ticket.java (hunk 18 above)
Comment (stasevich741) [reply src/main/java/cinema/entity/Ticket.java]: fixed

**Pull request merged**